pydrive2/fs/spec.py

```diff
@@ -542,6 +542,34 @@ def cp_file(self, lpath, rpath, **kwargs):
             buffer = io.BytesIO(stream.read())
             self.upload_fobj(buffer, rpath)
 
+    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+
+        if recursive:
+            raise NotImplementedError("Recursive move is not supported")
+        if maxdepth:
+            raise NotImplementedError("Max depth move is not supported")
+
+        file1_name = posixpath.basename(path1)
+        file2_name = posixpath.basename(path2)
+
+        file1_parent = self._parent(path1)
+        file2_parent = self._parent(path2)
+
+        file1_id = self._get_item_id(path1)
+
+        file1 = self.client.CreateFile({"id": file1_id})
+
+        file1.FetchMetadata(fields="title,parents")
```

Comment:
need to fix it eventually ...  I don't think we have to do this call

Reply:
To fix this we need a way to tell `_FilesUpdate` which fields to include in the body. 

Now this in done trough `param["body"] = self.GetChanges()`:

line 770:
https://github.com/iterative/PyDrive2/blob/64a50a1580b2d276f8c9563f5ba620f7e7706119/pydrive2/files.py#L761-L789

and `GetChanges` is:

https://github.com/iterative/PyDrive2/blob/64a50a1580b2d276f8c9563f5ba620f7e7706119/pydrive2/apiattr.py#L92-L103

And that returns a blank dict if the metadata was not fetched before.

I still don't understand this logic well but i think that when you do `FetchMetadata`, at the end calls:

https://github.com/iterative/PyDrive2/blob/64a50a1580b2d276f8c9563f5ba620f7e7706119/pydrive2/apiattr.py#L86-L90

that is the same dict used by `GetChanges`.

If there is a bug, it's there somewhere. i will try to play around and see what happen.

Reply:
@shcheklein what do you think? i can't dig deeper than that, i don't really know the internal logic

Reply:
@simone-viozzi hey, sorry, I haven't had time to get to this :( we got into a few areas that would require some brainstorming, so I need a bit more time. I'll try to get there soon.

Reply:
https://github.com/iterative/PyDrive2/pull/223 should fix this, please rebase and let's try to simplify this